### PR TITLE
Fix the metadata setup compatibility issue

### PR DESCRIPTION
**Pull request by zymap**
---

*Motivation*

#8269 change the arguments name which causes the setup
metadata command is not compatible with old versions.
For keeping the compatibility of the setup metadata command,
we should avoid deleting the existing arguments. If we need to
change it, it's better to keep the old arguments and mark it
deprecates and hide them.

### CODE REVIEW

**Review by eolivelli — COMMENTED** [review]
Can we add unit tests in order to prevent regressions?

File: pulsar-broker/src/main/java/org/apache/pulsar/PulsarClusterMetadataSetup.java [hunk 1]
```diff
@@ -115,6 +115,13 @@
                 description = "The metadata service URI of the existing BookKeeper cluster that you want to use")
         private String existingBkMetadataServiceUri;
```
Comment (eolivelli) [inline pulsar-broker/src/main/java/org/apache/pulsar/PulsarClusterMetadataSetup.java:117]: This annotation is not really meaningful here by itself, can we add a comment?

**Review by eolivelli — CHANGES_REQUESTED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/PulsarClusterMetadataSetup.java (hunk 1 above)
Comment (zymap) [reply pulsar-broker/src/main/java/org/apache/pulsar/PulsarClusterMetadataSetup.java:117]: Which annotation? Do you mean `@Deprecated`?

**Review by codelipenghui — APPROVED** [review]

**Review by sijie — APPROVED** [review]

**Comment by Jennifer88huang** [comment]
@zymap @sijie I don't think we need to add `doc-required` label for this PR. 
- In [PR-8269](https://github.com/apache/pulsar/pull/8269), we've added the usage for the new commands. - 
- In this PR, we add the deprecated commands back in code and mark them as deprecated. So these deprecated commands can be used by existing users, yet it's encouraged to be used nor displayed in the -help list. We don't need to add usage for those deprecated commands.
What do you think about it?

**Comment by zymap** [comment]
Yes. I think we can remove the `doc-required` label for this PR.

**Comment by Jennifer88huang** [comment]
@zymap thank you for your confirmation.

**Pull request merged**